File: problems/move_zeroes/MoveZeroes.java
package problems.move_zeroes;

public class MoveZeroes {
    public void moveZeroes(int[] nums) {
        int i = 0;
        int j = 0;

        while (j < nums.length) {
            if (nums[j] != 0) {
                // swap with i
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i += 1;
            }
            j += 1;
        }
    }
}

File: problems/move_zeroes/move_zeroes.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0
        right = 0
        while right < len(nums):
            if nums[right] != 0:
                temp = nums[left]
                nums[left] = nums[right]
                nums[right] = temp
                left += 1
            right += 1
